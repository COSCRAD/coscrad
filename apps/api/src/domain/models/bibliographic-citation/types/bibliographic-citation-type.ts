import { BibliographicCitationType } from '@coscrad/api-interfaces';

export { BibliographicCitationType };

export const isBibliographicCitationType = (input: unknown): input is BibliographicCitationType =>
    Object.values(BibliographicCitationType).includes(input as BibliographicCitationType);
