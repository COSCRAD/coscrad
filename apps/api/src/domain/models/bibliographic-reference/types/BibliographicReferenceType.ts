export enum BibliographicReferenceType {
    book = 'book',
}
