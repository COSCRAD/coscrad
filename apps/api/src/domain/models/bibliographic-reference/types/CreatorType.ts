export enum CreatorType {
    author = 'Author',
    director = 'Director',
    artist = 'Artist',
}
