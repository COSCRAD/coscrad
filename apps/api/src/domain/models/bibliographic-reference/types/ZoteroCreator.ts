type ZoteroHumanCreator = {
    firstName: string; // non-zero length

    lastName: string; // non-zero length

    type: CreatorType; // isEnum(CreatorType)
};

type ZoteroInstitutionCreator = {
    name: string; //non-zero length

    type: CreatorType; // isEnum(CreatorType);
};

export type ZoteroCreator = ZoteroHumanCreator | ZoteroInstitutionCreator;

export enum CreatorType {
    author = 'Author',
    artist = 'Artist',
    director = 'Director',
}
