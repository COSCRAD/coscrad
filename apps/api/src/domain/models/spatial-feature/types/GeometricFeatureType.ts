// Let's trying using a real enum and see if it causes any grief
export enum GeometricFeatureType {
    point = 'Point',
    line = 'LineString',
    polygon = 'Polygon',
}
