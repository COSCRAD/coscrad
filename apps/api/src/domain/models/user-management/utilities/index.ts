export * from './coscrad-date.entity';
export * from './invalid-date.error';
