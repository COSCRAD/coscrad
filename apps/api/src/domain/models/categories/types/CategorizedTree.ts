import { DTO } from '../../../../types/DTO';
import { Category } from '../entities/category.entity';

export type CategorizedTreeNode = DTO<Category>;

export type CategorizedTree = CategorizedTreeNode[];
