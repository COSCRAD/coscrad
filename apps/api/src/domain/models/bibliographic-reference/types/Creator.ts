import { DTO } from '../../../../types/DTO';
import { isNullOrUndefined } from '../../../utilities/validation/is-null-or-undefined';
import BaseDomainModel from '../../BaseDomainModel';
import { CreatorType } from './CreatorType';

interface IBibliographicReferenceDataCreator {
    name: string;
    type: CreatorType;
}

export default class Creator extends BaseDomainModel implements IBibliographicReferenceDataCreator {
    readonly name: string;

    readonly type: CreatorType;

    constructor(dto: DTO<Creator>) {
        super();

        if (isNullOrUndefined(dto)) return;

        this.name = dto.name;

        this.type = dto.type;
    }
}
