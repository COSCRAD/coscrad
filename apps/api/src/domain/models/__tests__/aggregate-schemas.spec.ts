import { getCoscradDataSchema } from '@coscrad/data-types';
import formatAggregateType from '../../../view-models/presentation/formatAggregateType';
import { getAggregateCtor } from '../../factories/utilities/getAggregateCtor';
import { isDiscriminatedUnionResourceType } from '../../factories/utilities/isDiscriminatedUnionResourceType';
import { AggregateType } from '../../types/AggregateType';
import { BibliographicReferenceType } from '../bibliographic-reference/types/BibliographicReferenceType';
import { GeometricFeatureType } from '../spatial-feature/types/GeometricFeatureType';

type AggregateTypeAndSubtype = [
    AggregateType,
    null | typeof BibliographicReferenceType | typeof GeometricFeatureType
];

describe(`Coscrad Data Schemas for aggregate root domain models`, () => {
    Object.values(AggregateType)
        .flatMap((aggregateType: AggregateType): AggregateTypeAndSubtype[] => {
            if (isDiscriminatedUnionResourceType(aggregateType)) {
                if (aggregateType === AggregateType.bibliographicReference)
                    return Object.values(BibliographicReferenceType).map(
                        (bibliographicReferenceType) => [aggregateType, bibliographicReferenceType]
                    ) as unknown as AggregateTypeAndSubtype[];

                if (aggregateType === AggregateType.spatialFeature)
                    return Object.values(GeometricFeatureType).map((geometricFeatureType) => [
                        aggregateType,
                        geometricFeatureType,
                    ]) as unknown as AggregateTypeAndSubtype[];
            }

            // Model has no `subtype` as it is not comprised of a discriminated union
            return [[aggregateType, null]];
        })
        .forEach(([aggregateType, subtype]) =>
            describe(`the COSCRAD data shcmea for a ${formatAggregateType(aggregateType)}`, () => {
                it('match the snapshot', () => {
                    const Ctor = getAggregateCtor(aggregateType, subtype as unknown as string);

                    const dataSchema = getCoscradDataSchema(Ctor);

                    expect(dataSchema).toMatchSnapshot();
                });
            })
        );
});
