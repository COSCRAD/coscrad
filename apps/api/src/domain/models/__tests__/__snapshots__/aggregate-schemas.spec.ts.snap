// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a BibliographicReference of subtype: book match the snapshot 1`] = `
Object {
  "data": Object {
    "coscradDataType": Object {
      "abstract": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "creators": Object {
        "coscradDataType": Object {
          "name": Object {
            "coscradDataType": "NON_EMPTY_STRING",
            "isArray": false,
            "isOptional": false,
          },
          "type": Object {
            "coscradDataType": "ENUM",
            "enumLabel": "Creator Type",
            "enumName": "BIBLIOGRAPHIC_SUBJECT_CREATOR_TYPE",
            "isArray": false,
            "isOptional": false,
            "labelsAndValues": Array [
              Object {
                "label": "artist",
                "value": "artist",
              },
              Object {
                "label": "author",
                "value": "author",
              },
              Object {
                "label": "director",
                "value": "director",
              },
            ],
          },
        },
        "isArray": true,
        "isOptional": false,
      },
      "isbn": Object {
        "coscradDataType": "ISBN",
        "isArray": false,
        "isOptional": true,
      },
      "numberOfPages": Object {
        "coscradDataType": "POSITIVE_INTEGER",
        "isArray": false,
        "isOptional": true,
      },
      "place": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "publisher": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "title": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
      "url": Object {
        "coscradDataType": "URL",
        "isArray": false,
        "isOptional": true,
      },
      "year": Object {
        "coscradDataType": "YEAR",
        "isArray": false,
        "isOptional": true,
      },
    },
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a BibliographicReference of subtype: courtCase match the snapshot 1`] = `
Object {
  "data": Object {
    "coscradDataType": Object {
      "abstract": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "caseName": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
      "court": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "dateDecided": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "pages": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "url": Object {
        "coscradDataType": "URL",
        "isArray": false,
        "isOptional": true,
      },
    },
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a BibliographicReference of subtype: journalArticle match the snapshot 1`] = `
Object {
  "data": Object {
    "coscradDataType": Object {
      "abstract": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "creators": Object {
        "coscradDataType": Object {
          "name": Object {
            "coscradDataType": "NON_EMPTY_STRING",
            "isArray": false,
            "isOptional": false,
          },
          "type": Object {
            "coscradDataType": "ENUM",
            "enumLabel": "Creator Type",
            "enumName": "BIBLIOGRAPHIC_SUBJECT_CREATOR_TYPE",
            "isArray": false,
            "isOptional": false,
            "labelsAndValues": Array [
              Object {
                "label": "artist",
                "value": "artist",
              },
              Object {
                "label": "author",
                "value": "author",
              },
              Object {
                "label": "director",
                "value": "director",
              },
            ],
          },
        },
        "isArray": true,
        "isOptional": false,
      },
      "doi": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "issn": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "issueDate": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
      "publicationTitle": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": true,
      },
      "title": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
      "url": Object {
        "coscradDataType": "URL",
        "isArray": false,
        "isOptional": true,
      },
    },
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Book  match the snapshot 1`] = `
Object {
  "author": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "pages": Object {
    "coscradDataType": Object {
      "identifier": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
      "text": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
      "translation": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
    },
    "isArray": false,
    "isOptional": false,
  },
  "publicationDate": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "subtitle": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "title": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Category  match the snapshot 1`] = `
Object {
  "childrenIDs": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": true,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "label": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Media Item  match the snapshot 1`] = `
Object {
  "contributorAndRoles": Object {
    "coscradDataType": Object {
      "contributorId": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
      "role": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
    },
    "isArray": true,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "mimeType": Object {
    "coscradDataType": "ENUM",
    "enumLabel": "MIME Type",
    "enumName": "MIME_TYPE",
    "isArray": false,
    "isOptional": false,
    "labelsAndValues": Array [
      Object {
        "label": "mp3",
        "value": "audio/mpeg",
      },
      Object {
        "label": "mp4",
        "value": "video/mp4",
      },
    ],
  },
  "title": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "titleEnglish": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "url": Object {
    "coscradDataType": "URL",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Note  match the snapshot 1`] = `
Object {
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "note": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Photograph  match the snapshot 1`] = `
Object {
  "dimensions": Object {
    "coscradDataType": Object {
      "heightPX": Object {
        "coscradDataType": "NON_NEGATIVE_FINITE_NUMBER",
        "isArray": false,
        "isOptional": false,
      },
      "widthPX": Object {
        "coscradDataType": "NON_NEGATIVE_FINITE_NUMBER",
        "isArray": false,
        "isOptional": false,
      },
    },
    "isArray": false,
    "isOptional": false,
  },
  "filename": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "photographer": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Song  match the snapshot 1`] = `
Object {
  "audioURL": Object {
    "coscradDataType": "URL",
    "isArray": false,
    "isOptional": false,
  },
  "contributions": Object {
    "coscradDataType": Object {
      "contributorId": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
      "role": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
    },
    "isArray": true,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "lengthMilliseconds": Object {
    "coscradDataType": "NON_NEGATIVE_FINITE_NUMBER",
    "isArray": false,
    "isOptional": false,
  },
  "lyrics": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "startMilliseconds": Object {
    "coscradDataType": "NON_NEGATIVE_FINITE_NUMBER",
    "isArray": false,
    "isOptional": false,
  },
  "title": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "titleEnglish": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Spatial Feature of subtype: LineString match the snapshot 1`] = `
Object {
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Spatial Feature of subtype: Point match the snapshot 1`] = `
Object {
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Spatial Feature of subtype: Polygon match the snapshot 1`] = `
Object {
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Tag  match the snapshot 1`] = `
Object {
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "label": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "members": Object {
    "coscradDataType": "COMPOSITE_IDENTIFIER",
    "isArray": true,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Term  match the snapshot 1`] = `
Object {
  "audioFilename": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "contributorId": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "sourceProject": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "term": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "termEnglish": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Transcribed Audio Item  match the snapshot 1`] = `
Object {
  "audioFilename": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "lengthMilliseconds": Object {
    "coscradDataType": "NON_NEGATIVE_FINITE_NUMBER",
    "isArray": false,
    "isOptional": false,
  },
  "startMilliseconds": Object {
    "coscradDataType": "NON_NEGATIVE_FINITE_NUMBER",
    "isArray": false,
    "isOptional": false,
  },
  "transcript": Object {
    "coscradDataType": Object {
      "timeRanges": Object {
        "coscradDataType": Object {
          "data": Object {
            "coscradDataType": "NON_EMPTY_STRING",
            "isArray": false,
            "isOptional": true,
          },
          "inPoint": Object {
            "coscradDataType": "NON_NEGATIVE_FINITE_NUMBER",
            "isArray": false,
            "isOptional": false,
          },
          "outPoint": Object {
            "coscradDataType": "NON_NEGATIVE_FINITE_NUMBER",
            "isArray": false,
            "isOptional": false,
          },
        },
        "isArray": true,
        "isOptional": false,
      },
    },
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a User  match the snapshot 1`] = `
Object {
  "authProviderUserId": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "profile": Object {
    "coscradDataType": Object {
      "email": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
      "name": Object {
        "coscradDataType": Object {
          "firstName": Object {
            "coscradDataType": "NON_EMPTY_STRING",
            "isArray": false,
            "isOptional": false,
          },
          "lastName": Object {
            "coscradDataType": "NON_EMPTY_STRING",
            "isArray": false,
            "isOptional": false,
          },
        },
        "isArray": false,
        "isOptional": false,
      },
    },
    "isArray": false,
    "isOptional": true,
  },
  "roles": Object {
    "coscradDataType": "ENUM",
    "enumLabel": "User Role",
    "enumName": "COSCRAD_USER_ROLE",
    "isArray": true,
    "isOptional": false,
    "labelsAndValues": Array [
      Object {
        "label": "admin",
        "value": "projectAdmin",
      },
      Object {
        "label": "viewer",
        "value": "viewer",
      },
      Object {
        "label": "COSCRAD admin",
        "value": "superAdmin",
      },
    ],
  },
  "username": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a User Group  match the snapshot 1`] = `
Object {
  "description": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "label": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "userIds": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": true,
    "isOptional": false,
  },
}
`;

exports[`Coscrad Data Schemas for aggregate root domain models the COSCRAD data schema for a Vocabulary List  match the snapshot 1`] = `
Object {
  "entries": Object {
    "coscradDataType": Object {
      "termId": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
    },
    "isArray": false,
    "isOptional": false,
  },
  "id": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": false,
  },
  "name": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "nameEnglish": Object {
    "coscradDataType": "NON_EMPTY_STRING",
    "isArray": false,
    "isOptional": true,
  },
  "variables": Object {
    "coscradDataType": Object {
      "name": Object {
        "coscradDataType": "NON_EMPTY_STRING",
        "isArray": false,
        "isOptional": false,
      },
    },
    "isArray": true,
    "isOptional": false,
  },
}
`;
